frontend/src/pages/groups/GroupForm.tsx

```diff
@@ -0,0 +1,126 @@
+import { FunctionComponent, ReactNode, useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+import { Group, GroupInput, GroupType } from '../../types/api-types'
+import TextField from '../../components/forms/TextField'
+import SelectField from '../../components/forms/SelectField'
+import Button from '../../components/Button'
+
+interface Props {
+  /**
+   * If true, the Submit button will be disabled
+   */
+  isLoading: boolean
+  /**
+   * The text to display in the Submit button of the form
+   */
+  submitButtonLabel: ReactNode
+  /**
+   * The values to display in the fields of the form. Note that this is NOT a
+   * controlled component.
+   */
+  defaultValues?: Group
+  /**
+   * The callback triggered when the user submits the form
+   */
+  onSubmit: (input: GroupInput) => void
+}
+
+/**
+ * This component encapsulates a form for creating and editing groups.
+ */
+const GroupForm: FunctionComponent<Props> = (props) => {
+  const { register, handleSubmit, reset } = useForm()
```

Comment:
I'm extremely happy with this library. It saved me a ton of manual work while letting me handle the styling however I want.

Reply:
Yeah this looks pretty slick!